cmd/stolonctl/status.go

```diff
@@ -135,3 +159,84 @@ func translateBoolean(value bool) string {
 	}
 	return "No"
 }
+
+func isClusterHealthy(status *cluster.Status) (unhealthyReason string, healthy bool) {
```

Comment:
What am I expecting descending into this function? A overview of what the cluster status is made of. Contrast to this:
```go
func isClusterHealthy(status *cluster.Status) (reasonUnhealthy string, healthy bool) {
	metrics := collectMetricsFromStatus(status)
	reasonUnhealthy = computeReasonUnhealthy(metrics)
	if reasonUnhealthy != "" {
		return reasonUnhealthy, false
	}
	return "", true
}

func collectMetricsFromStatus(status *cluster.Status) (result statusMetrics) {
	var masterID string
	for _, pod := range status.PodsStatus {
		var keeperID string
		for _, keeperState := range status.ClusterData.KeepersState {
			if pod.PodIP == keeperState.ListenAddress {
				keeperID = keeperState.ID
			}
		}
		if keeperID != "" && status.ClusterData.KeepersState[keeperID].PGState != nil {
			result.runningPods++
			if status.ClusterData.KeepersState[keeperID].PGState.Role == common.MasterRole {
				if status.ClusterData.KeepersState[keeperID].Healthy {
					result.masterHealthy = true
					masterID = keeperID
				}
				// count amount of running master nodes
				result.runningMasters++
			} else {
				// count amount of running standby nodes
				result.runningStandbys++
				if masterID != "" {
					if status.ClusterData.KeepersState[masterID].PGState.XLogPos > status.ClusterData.KeepersState[keeperID].PGState.XLogPos {
						if status.ClusterData.KeepersState[masterID].PGState.XLogPos-status.ClusterData.KeepersState[keeperID].PGState.XLogPos > 0 {
							result.replicationLag = true
						}
					} else {
						if status.ClusterData.KeepersState[keeperID].PGState.XLogPos-status.ClusterData.KeepersState[masterID].PGState.XLogPos > 0 {
							result.replicationLag = true
						}
					}
				}
			}
			if status.ClusterData.KeepersState[keeperID].Healthy {
				// count amount of running healthy nodes
				result.healthyNodes++
			}
		}
	}
	return result
}

func computeReasonUnhealthy(metrics statusMetrics) (reason string) {
	if metrics.runningPods <= 1 {
		return "cluster is running with less than 2 nodes"
	}
	if metrics.runningMasters == 0 {
		return "cluster has no master node"
	}
	if metrics.runningMasters > 1 {
		return "cluster has more than one master node"
	}
	if !metrics.masterHealthy {
		return "master is unhealthy"
	}
	if metrics.runningStandbys == 0 {
		return "cluster has no standby servers"
	}
	if metrics.healthyNodes < 2 {
		return fmt.Sprintf("there are only %v healthy keeper nodes in cluster", metrics.healthyNodes)
	}
	if metrics.replicationLag {
		return "high replication lag between master and stanby(s)"
	}
	return ""
}
```